### PR TITLE
Add CI configuration for Rook-Ceph operator

### PR DESCRIPTION
**Pull request by leseb**
Adding the openshift/rook repo.

Signed-off-by: Sébastien Han <seb@redhat.com>

@umangachapagain @travisn @BlaineEXE @obnoxxx FYI

### CODE REVIEW

File: ci-operator/config/openshift/rook/openshift-rook-master.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+binary_build_commands: GOPATH=$(go env GOPATH) make clean && make -j$nproc IMAGES='ceph' VERSION=0 build
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.15
```
Comment (umangachapagain) [inline ci-operator/config/openshift/rook/openshift-rook-master.yaml:6]:
```suggestion
    tag: rhel-8-release-golang-1.15-openshift-4.6
```
maybe this has the `shasum` tool required for builds.

File: ci-operator/config/openshift/rook/openshift-rook-master.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+test_binary_build_commands: GOPATH=$(go env GOPATH) make -j $(nproc) test
+tests:
+- artifact_dir: /tmp/artifacts
+  as: unit
+  commands: GOPATH=$(go env GOPATH) make -j $(nproc) test
```
Comment (droslean) [inline ci-operator/config/openshift/rook/openshift-rook-master.yaml:18]: why you need `GOPATH=$(go env GOPATH)` in every command?

Comment (leseb) [reply ci-operator/config/openshift/rook/openshift-rook-master.yaml:18]: This is coming straight from our existing CI pipeline, perhaps it's not needed but I know nothing about this CI so I thought I'd just duplicate what we already do.

**Comment by leseb** [comment]
Waiting for https://github.com/rook/rook/pull/7125 so we can move forward...

**Review by umangachapagain — APPROVED** [review]

**Comment by leseb** [comment]
@bbguimaraes how can we move forward with that? Thanks.

**Pull request merged**